src/platform_linux/shmem.c

```diff
@@ -581,6 +581,16 @@ platform_shm_realloc(platform_heap_id hid,
                      const char      *file,
                      const int        lineno)
 {
+   debug_assert(((oldptr == NULL) && (oldsize == 0)) || (oldptr && oldsize),
+                "oldptr=%p, oldsize=%lu",
+                oldptr,
+                oldsize);
+
+   // We can only realloc from an oldptr that's allocated from shmem
+   debug_assert(!oldptr || platform_valid_addr_in_heap(hid, oldptr),
```

Comment:
Assertions added just to ensure that no mistakes are made in calling conventions (in future).